packages/m2m-gateway/src/services/agreementService.ts

```diff
@@ -1,8 +1,43 @@
+import { WithLogger } from "pagopa-interop-commons";
+import { m2mGatewayApi } from "pagopa-interop-api-clients";
 import { PagoPAInteropBeClients } from "../clients/clientsProvider.js";
+import { M2MGatewayAppContext } from "../utils/context.js";
+import {
+  toGetAgreementsApiQueryParams,
+  toM2MGatewayApiAgreement,
+} from "../api/agreementApiConverter.js";
 
 export type AgreementService = ReturnType<typeof agreementServiceBuilder>;
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function agreementServiceBuilder(_clients: PagoPAInteropBeClients) {
-  return {};
+export function agreementServiceBuilder(clients: PagoPAInteropBeClients) {
+  return {
+    getAgreements: async (
+      queryParams: m2mGatewayApi.GetAgreementsQueryParams,
+      ctx: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.Agreements> => {
```

Comment:
```suggestion
    async getAgreements (
      queryParams: m2mGatewayApi.GetAgreementsQueryParams,
      ctx: WithLogger<M2MGatewayAppContext>
    ): Promise<m2mGatewayApi.Agreements> {
```

As @ecamellini was suggesting [here](https://github.com/pagopa/interop-be-monorepo/pull/1822#discussion_r2084684352), just for consistency with the other services

Reply:
MMh, we could really use the help of a linter here